static/app.js

```diff
@@ -42,6 +41,46 @@ $(function() {
         console.log('Geolocation not enabled in this browser')
     }
 
+    let timeoutId;
+
+    $('#term').on('keyup', ()=> {
+        if ($('#search_type').val() == 'by keyword') {
+            clearTimeout(timeoutId);
+            timeoutId = setTimeout(runAutocomplete, 500)
+        } 
+    });
+
+    async function runAutocomplete(term) {
+        $('#auto').remove()
+        term = $('#term').val()
+        try {
+            const response = await axios({
+                url: `${base_url}autocomplete`,
+                method: "GET",
+                params: {
+                    query: term
+                }
+            })
+            $('#term').after('<div id="auto"></div>')
+            $('#auto').on('click', grabItem)
+            Object.entries(response.data).forEach((entry) => {
+                const [key, value] = entry;
+                const brewery = value
+                renderAutoComplete(brewery)
```

Comment:
In this case you're re-rendering in a loop. Which could be laggy in some browsers. Can we avoid this?

Could you possibly get the data and do it all in one render?

Reply:
I found a way to avoid this issue and instead break down into several functions.  In the new iteration, I'm using the processBreweries function for processing the brewery data into an array (breweriesResults), and then appending to the DOM in one go.  

/** helper function to process brewery API response data
     * takes response data and dataType as parameters, turns response data into an results object to be rendered
     * dataType is in regards to autocomplete or normal results
     */

function processBreweries(data, dataType){
        const breweriesResults = Object.entries(data).map((entry) => {
            const [key, value] = entry;
            const brewery = value
            return createBrewery(brewery, dataType)
        })
        return breweriesResults;
    }

 /** helper function that takes a brewery object as a parameter, creates and then returns a brewery DOM element. */

    function createBrewery(brewery, dataType) {
        let breweryElement;
        if (dataType == 'autoComplete') {
            breweryElement = $(`<p>${brewery.name}</p>`)
        }
        else {
             breweryElement = `<h5><a href='/breweries/${brewery.id}'>${brewery.name}</a></h5>
                <p>${brewery.city}, ${brewery.state}</p>`
            }
        return breweryElement
    }

/** helper function that appends the breweries data to a given element in the DOM */

    function renderBreweries(breweries, element) {
        $(element).empty().append(breweries)
    }